servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -244,7 +267,184 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
+                    return rc.executionContext();
+                }
+
+                @Override
+                public StreamingHttpResponseFactory httpResponseFactory() {
+                    return rc.httpResponseFactory();
+                }
+
+                @Override
+                public Completable onClose() {
+                    return rc.onClose();
+                }
+
+                @Override
+                public Completable closeAsync() {
+                    return rc.closeAsync();
+                }
+
+                @Override
+                public Completable closeAsyncGracefully() {
+                    return rc.closeAsyncGracefully();
+                }
+
+                @Override
+                public StreamingHttpRequest newRequest(final HttpRequestMethod method, final String requestTarget) {
+                    return rc.newRequest(method, requestTarget);
+                }
+            }).shareContextOnSubscribe();
+        });
+        }
+
+        @Override
+        public HttpClient asClient() {
+            return toClient(this, singleClient.executionContext().executionStrategy());
+        }
+
+        @Override
+        public BlockingStreamingHttpClient asBlockingStreamingClient() {
+            return toBlockingStreamingClient(this, singleClient.executionContext().executionStrategy());
+        }
+
+        @Override
+        public BlockingHttpClient asBlockingClient() {
+            return toBlockingClient(this, singleClient.executionContext().executionStrategy());
+        }
+
+        @Override
+        public StreamingHttpRequest newRequest(final HttpRequestMethod method, final String requestTarget) {
+            return singleClient.newRequest(method, requestTarget);
+        }
+
+        @Override
+        public Single<StreamingHttpResponse> request(final StreamingHttpRequest request) {
+            return defer(() -> {
+                applyMultiAddressStrategy(request.context(), singleClient.executionContext().executionStrategy());
+                return singleClient.request(request);
+            });
+        }
+
+        private HttpExecutionStrategy applyMultiAddressStrategy(final ContextMap contextMap,
+                                                                final HttpExecutionStrategy singleClientStrategy) {
+            HttpExecutionStrategy requestStrategy = contextMap.get(HTTP_EXECUTION_STRATEGY_KEY);
+            HttpApiConversions.ClientAPI clientAPI = contextMap.get(HTTP_CLIENT_API_KEY);
+            HttpExecutionStrategy useStrategy = requestStrategy.hasOffloads() ?
+                    // multi-client offloads something
+                    defaultStrategy() == singleClientStrategy ?
+                        defaultStrategy() == requestStrategy ?
+                                // async streaming client with default strategy
+                                HttpApiConversions.ClientAPI.ASYNC_STREAMING.defaultStrategy() :
```

Comment:
Last time we discussed this problem, we agreed to keep it simple: take whatever `HTTP_EXECUTION_STRATEGY_KEY` comes from multi-address level and merge it with a `singleClientStrategy` to make sure we do not miss a case when single-address client requires more offloading. We agreed to avoid an extra key, like `HTTP_CLIENT_API_KEY`, and any dependency on API variant. Can you clarify why this was necessary to consider API at the end?